documentation/adr/0022-use-keyword-args.md

```diff
@@ -0,0 +1,92 @@
+This document reviews this in detail, and recommends a convention for function arguments.
+
+
+### Python
+
+Currently, we don’t have an established syntax for Python. Because we have a clean slate it makes sense to only have a Pythonic syntax using required positional, optional positional, and optional keyword/value pairs
+
+```python
+def function_name(a, b, *args, **kwargs):
+	# body
+
+function_name(a, b, c, d, keyword_1=val_1, keyword_2=val_2, …)
+```
+Any argument may be specified as `value` or `name=value`.
+All arguments to the right of an argument which has been provided as a name/value pair must follow that syntax.
+The order of named arguments is not fixed.
+
+Python does not natively support flag arguments, and they are not used in the core language. They are supported in a small number of Python libraries, notably `matplotlib` that was modelled on the MATLAB API. Many of these have been removed in more recent releases of the library. Flag arguments may be implemented on any function as optional string arguments with custom handling.
+
+
+### Matlab
+
+The Horace/Herbert source uses a wide range of API syntax:
+
+- `flagname` and `-flagname` flag arguments
+- true and false flags: `flagname` (`true`) and `noflagname` (`false`)
+- key-value pairs `'keyword', value, ...`
+- positional parameters
+- optional parameters interspersed with positional parameters in some functions
+
+The key-value pair syntax is equivalent to the Pythonic syntax, e.g. `function_name(a, b, c, d, ‘keywordA’, valA, ‘keywordB’, true, …)`.
+
+The mix of `-flagname` and `flagname` introduces complexity into argument parsing - specifically identifying if a string argument `flagname` a flag or a keyword argument?
+
+Inclusion of negated flags allows inconsistent, ambiguous or multiplicitous arguments to be defined, complicating the user experience, e.g. `-disable-x` vs `-nodisable-x` vs `-enable-x` vs `-noenable-x` vs `-x`
+
+## Decision
+
+The MATLAB APIs must support positional, optional flag and keyword arguments:
+
+- flag arguments will be prefixed with `-`
+- flag arguments will default to `false` if not passed
+- if the same parameter is passed as both a flag and keyword argument with inconsistent values an error will be raised, otherwise the parameter is set to the passed value
+- support for negated flags, prefixed with `no`, will be removed. All flags will map to `true` when specified. Flag names may include the `no` prefix to make their meaning clear, e.g. `-nopix` will suppress pixel handling in that function
+- flag arguments may be truncated to the minimum unambiguous string for all flags defined on that function e.g. `-flagname`, `-flag`, `-f` are all valid. No other abbreviations will be supported
+- keyword arguments may not be abbreviated.
+
+The Python API will use ordered (mandatory) positional, optional positional and keyword arguments in the function definition.
+
+All optional arguments must appear between the positional and keyword arguments.
```

Comment:
This is a feature of the `inputParser` rather than the native named arg vs varargin syntax in MATLAB.

I disagree that it means that optional positional arguments aren't required to appear before keyword arguments, it just means that optional positional arguments may also be passed using keyword syntax.